src/parser/type_checking.c

```diff
@@ -18237,24 +18237,37 @@ pt_fold_const_expr (PARSER_CONTEXT * parser, PT_NODE * expr, void *arg)
     }
 
   if (prm_get_bool_value (PRM_ID_ORACLE_STYLE_EMPTY_STRING)
-      && (op == PT_STRCAT || op == PT_PLUS || op == PT_CONCAT || op == PT_CONCAT_WS))
+      && (op == PT_STRCAT || op == PT_PLUS || op == PT_CONCAT || op == PT_CONCAT_WS || op == PT_REPLACE))
     {
       TP_DOMAIN *domain;
 
+      if (op == PT_REPLACE)
+	{
+	  opd3 = expr->info.expr.arg3;
+	  type3 = (opd3) ? opd3->type_enum : PT_TYPE_NULL;
+	}
+
       /* use the caching variant of this function ! */
       domain = pt_xasl_node_to_domain (parser, expr);
 
       if (domain
 	  && (QSTR_IS_ANY_CHAR_OR_BIT (TP_DOMAIN_TYPE (domain)) || type1 == PT_TYPE_NULL || type2 == PT_TYPE_NULL))
 	{
-	  if (opd1 && opd1->node_type == PT_VALUE && type1 == PT_TYPE_NULL)
+	  if (opd3 && opd3->node_type == PT_VALUE && type3 == PT_TYPE_NULL)	/* REPLACE */
```

Comment:
Adding this conditional block results in 
```
csql> select replace('abc', 'a', null);

=== <Result of SELECT Command in Line 1> ===

   replace('abc', 'a', null)
======================
  'abc'
```
while the current result is 
```
csql> select replace('abc', 'a', null);

=== <Result of SELECT Command in Line 1> ===

   replace('abc', 'a', null)
======================
  NULL                
```
Note the Oracle's result of 
```
SELECT replace('abc', 'a', null) FROM dual;
```
is 'bc'.